src/DotLiquid.Tests/Filters/StandardFiltersV20Tests.cs

```diff
@@ -100,5 +97,195 @@ public void TestReplaceFirstRegexWorks()
         {
             Assert.That(ReplaceFirst(input: "a A A a", @string: "[Aa]", replacement: "b"), Is.EqualTo(expected: "b A A a"));
         }
+
+        [Test]
+        public void TestRoundTypes()
+        {
+            Assert.That(Round(1.2345678f, 2.0), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678, 2.0), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678m, 2.0), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+
+            Assert.That(Round(1.2345678f, 2.0f), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678, 2.0f), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678m, 2.0f), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+
+            Assert.That(Round(1.2345678f, 2.0m), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678, 2.0m), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678m, 2.0m), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+
+            Assert.That(Round(1.2345678f, 2), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678, 2), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+            Assert.That(Round(1.2345678m, 2), Is.EqualTo(1.23).And.TypeOf(typeof(decimal)));
+        }
+
+        [Test]
+        public void TestRoundHandlesBadParams()
+        {
+            Assert.That(Round("1.2345678", "two"), Is.Null);
+            Assert.That(Round("1.2345678", -2), Is.Null);
+            Assert.That(Round(1.123456789012345678901234567890123m, 50), Is.Null);
+
+            Assert.That(Round("1.2345678", 2.7), Is.EqualTo(1.235m));
+
+            Helper.AssertTemplateResult("1.235", "{{ 1.234678 | round: 2.7 }}", syntax: SyntaxCompatibilityLevel);
+            Helper.AssertTemplateResult("1.235", "{{ 1.234678 | round: 3.1 }}", syntax: SyntaxCompatibilityLevel);
+
+            Helper.AssertTemplateResult("", "{{ 1.234678 | round: -3 }}", syntax: SyntaxCompatibilityLevel);
+        }
+
+        [Test]
+        public void TestAtLeastTypes()
+        {
+            Assert.That(AtLeast(5, 5), Is.EqualTo(5).And.TypeOf(typeof(double)));
```

Comment:
Why are simple integer inputs resulting into a floating type (double)?

Reply:
This should be fixed now.